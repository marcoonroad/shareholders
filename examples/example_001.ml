module Sh = Shareholders

let secret = "If this is really secret, it must not be committed on Git."

let threshold, amount = (100, 100)

let checksum, shares = Sh.share ~secret ~threshold ~amount

let secret' = Sh.recover ~checksum ~shares

let _ =
  assert (secret = secret') ;
  print_endline checksum
