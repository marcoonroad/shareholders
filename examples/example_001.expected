ZC5JNmdLK05MZXdzc3NOUEpYcTFtckhpMUkyWU50UzJ5YTFmdmRWYkNRaFFxU0hpa25ITDNTQWVKd3JPZ0N3Z3h1Y0xxVi9NV1pmbHJrQUhrelpzUDFBdz09
